service/src/test/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsServiceTest.java

```diff
@@ -108,6 +112,91 @@ void setUp() {
         when(time.now()).thenReturn(NOW);
     }
 
+    @Nested
+    class PopulatePastAndFutureHearingListsTest {
+        @Test
+        void shouldPopulateAllHearingDateListsWhenPastAndFutureHearingsExist() {
+            Element<HearingBooking> futureHearing1 = hearingFromToday(3);
+            Element<HearingBooking> futureHearing2 = hearingFromToday(2);
+            Element<HearingBooking> todayHearing = hearingFromToday(0);
+            Element<HearingBooking> pastHearing1 = hearingFromToday(-2);
+            Element<HearingBooking> pastHearing2 = hearingFromToday(-3);
+
+            Object expectedHearingList = asDynamicList(List.of(
+                futureHearing1, futureHearing2), HearingBooking::toLabel);
+            Object expectedPastHearingList = asDynamicList(List.of(
+                todayHearing, pastHearing1, pastHearing2), HearingBooking::toLabel);
+            Object expectedFutureHearingList = asDynamicList(List.of(
+                futureHearing1, futureHearing2, todayHearing), HearingBooking::toLabel);
+
+            CaseData initialCaseData = CaseData.builder()
+                .hearingDetails(List.of(futureHearing1, futureHearing2, todayHearing, pastHearing1, pastHearing2))
+                .build();
+
+            Map<String, Object> data = service.populatePastAndFutureHearingLists(initialCaseData);
+
+            assertThat(data)
```

Comment:
i would prefer

            assertThat(data)
                .containsEntry(HAS_PAST_HEARING_FLAG, "Yes")
                .containsEntry(HAS_FUTURE_HEARING_FLAG, "Yes")
                .containsEntry(HAS_FUTURE_AND_TODAY_HEARING_FLAG, "Yes")
                .containsEntry(HAS_EXISTING_HEARINGS_FLAG, "Yes")
                .containsEntry(HEARING_DATE_LIST, expectedHearingList)
                .containsEntry(PAST_HEARING_LIST, expectedPastHearingList)
                .containsEntry(FUTURE_HEARING_LIST, expectedFutureHearingList);

as now it is really hard to match key with value